recipes/mongo-c-driver/all/conanfile.py

```diff
@@ -0,0 +1,155 @@
+            source_folder=self._source_subfolder,
+            build_folder=self._build_subfolder
+        )
+
+        return self._cmake
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = "{name}-{version}".format(
+            name=self.name,
+            version=self.version
+        )
+        os.rename(extracted_dir, self._source_subfolder)
+        # todo: fix it https://github.com/bincrafters/community/issues/995
+        tools.replace_in_file(
+            os.path.join(self._source_subfolder, "src", "libmongoc", "CMakeLists.txt"),
+            "add_executable (mongoc-stat ${PROJECT_SOURCE_DIR}/../../src/tools/mongoc-stat.c)",
+            "# add_executable (mongoc-stat ${PROJECT_SOURCE_DIR}/../../src/tools/mongoc-stat.c)")
+        tools.replace_in_file(
+            os.path.join(self._source_subfolder, "src", "libmongoc", "CMakeLists.txt"),
+            "target_link_libraries (mongoc-stat mongoc_shared ${LIBRARIES})",
+            "# target_link_libraries (mongoc-stat mongoc_shared ${LIBRARIES})")
+        if self.options.ssl_provider == 'LIBRESSL':
+            tools.replace_in_file(
+                os.path.join(self._source_subfolder, "src", "libmongoc", "CMakeLists.txt"),
+                self._libressl_find_pattern,
+                self._libressl_replacement_pattern.format(
+                    LIBRESSL_INCLUDE_DIRS=self.deps_cpp_info["libressl"].include_paths[0],
+                    LIBRESSL_LIBRARY_DIRS=self.deps_cpp_info["libressl"].lib_paths[0]
+                ))
+
+    def build(self):
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+        if self.options.shared:
+            del self.options.fPIC
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def requirements(self):
+        if self.options.ssl_provider == 'OPENSSL':
+            self.requires("openssl/1.1.1g")
+        if self.options.ssl_provider == 'LIBRESSL':
+            self.output.warn("Can be broken. Prefer OpenSSL instead of LIBRESSL")
+            self.requires("libressl/3.2.0")
+        if self.options.zlib == 'SYSTEM':
+            self.requires("zlib/1.2.11")
+            if tools.os_info.is_windows:
+                self.output.warn("Usage zlib provided by conan on Windows can be broken")
+
+    def package(self):
+        self.copy(pattern="COPYING", dst="licenses", src=self._source_subfolder)
+        self.copy(pattern="THIRD_PARTY_NOTICES", dst="licenses", src=self._source_subfolder)
+
+        cmake = self._configure_cmake()
+        cmake.install()
+        tools.rmdir(os.path.join(self.package_folder, "share"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+
+    def package_info(self):
```

Comment:
Fur future reference, we will need to generate the cmake components

https://github.com/mongodb/mongo-c-driver/blob/6fa33f0b1f1c219e6e943061a21e391422f56d48/src/libbson/examples/cmake/find_package/CMakeLists.txt#L35

Reply:
addressed in n-bes#1